src/lumigo_tracer/lumigo_utils.py

```diff
@@ -106,6 +106,16 @@ def get_region() -> str:
     pass
 
 
+class InternalState:
+    timeout_on_connection = False
+    internal_error_already_logged = False
+
+    @staticmethod
+    def reset():
```

Comment:
You call `reset` only in conftest. Don't you need to call it also at the beginning of the tracer execution? Won't it leak when there's a warm start?

Reply:
It should leak. This is a global state that we should have :)

Reply:
E.g. - when there is a timeout to the edge, don't waste more time on it (even in future invocations)